pkg/manager/impl/execution_manager.go

```diff
@@ -585,6 +585,7 @@ func (m *ExecutionManager) launchSingleTaskExecution(
 		EventVersion:        m.config.ApplicationConfiguration().GetTopLevelConfig().EventVersion,
 		RoleNameKey:         m.config.ApplicationConfiguration().GetTopLevelConfig().RoleNameKey,
 		RawOutputDataConfig: rawOutputDataConfig,
+		ClusterAssignment:   request.Spec.ClusterAssignment,
```

Comment:
Can we open an issue (or a follow up PR?) to make sure we resolve this against the matchable entity?